### ISSUE TITLE
Attiny85 + RF + sensor temperature

### ISSUE THREAD

**Issue by rocapal** [Kisaua/Attiny85-ds18b20-rf433-sensor#1]
Hi!

I was checking your tutorial here [1] which is very good! (english translation).

I could configure everything properly and I can program the ATtiny with the RBPi. The only problem I have is that I don't receive anything in the  RBPi with the binary 'rf433receive'. I have checked that the AT is transmitting something using a RTL-SDR dongle, so I can see that something is sent using 433MHz. But the problem is that rf433receive doesn't receive anything.

Any idea how I can debug and fix the problem?
Thanks!

Best,

[1] http://geek-mag.com/posts/255446/

**Comment by Kisaua** [comment]
Try to run simplesniff and try to see the message from AT or try uncomment the lines

//void log(string a){ 
 // cout << a ; 
 //} 
and other lines with log and cout to see messages.

**Comment by rocapal** [comment]
Hi @Kisaua , thanks for your answer!

I don't get it to work. I did enable the logs but nothing happens. Look at the trace (it doesn't go more than lock started) Even I can see this traces sometimes and the transmitter is not sensing anything.


`$ sudo ./rf433recieve 

Lock started
22822
Lock started
22997
Lock started
22529
Lock started
22446
Lock started
22719
Lock started
22403
Lock started
23266
`

However, I have seen that simpleSniff and rf433recieve apps have different values in the IF-control structure to check what's the duration. Is that normal? Which of the two apps is using the correct values?

You can check the screenshot how I can detect the signal sent by the AT with the RTL-SDR device.
![captura de pantalla de 2016-10-29 20-02-11](https://cloud.githubusercontent.com/assets/158031/19831902/a46e0af6-9e15-11e6-9be0-1a9c6aa9740f.png)


I'm also using the RPi-3 4-core, this is not relevant right?
Thanks for your help!

**Comment by Kisaua** [comment]
Yes, simpleSniff i have used as an example for my app rf433recieve.  For this reason there is different duration. 
in my app lock signai is 4 different durations in range
22000 - 24000
7000 - 9000
3500 - 4500 
and 1100 - 1500 
if app recieve this 4 durations then start recieve information bits. 
0 - logical state changes with the duration in range 600 - 950 and 650 -950
1 -  logical state changes with the duration in range 600 - 950 and 1500 -1850

In your case you see only the first duration in range 22000 -24000us. Try to change duration time from 6000 - 10000 instad of 7000 - 9000 and see Lock information.

Try to play with the durations. 
Here is more detatiled information about rf433 sniffing 

http://www.homautomation.org/2014/04/25/how-to-decode-x10-rf-protocol/

**Comment by rocapal** [comment]
I have tried to adapt the values of the locks but at the end I got an error when it decodes. What I don't get it is why my parameters are going a be different than yours. We are using the same hardware. Or this means that for every pair of RF (receiver and transmitter) I need to calibrate them to achieve the proper duration of each lock?

**Comment by Kisaua** [comment]
try to run 
https://github.com/vdemay/433Rpi/blob/master/searchLock.cpp
change the duration  in the line 54 from 11000 to 20000 to see the signal you have and then you can difine your durations. Why it is differnt from my values... ? no ideas.
